### ISSUE TITLE
Compiler bug trigged on OSX

### ISSUE THREAD

**Issue by bvssvni** [tomaka/glutin#359]
`rustc 1.0.0-nightly (f207ecbe0 2015-04-03) (built 2015-04-03)`

```
/Users/sven/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.1.1/src/declare.rs:138:21: 138:35 error: internal compiler error: coherence failed to report ambiguity: cannot locate the impl of the trait `core::marker::PhantomFn<<F as declare::IntoMethodImp>::Ret>` for the type `<F as declare::IntoMethodImp>::Ret`
/Users/sven/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.1.1/src/declare.rs:138     let mut types = F::Ret::encode().as_str().to_string();
                                                                                                              ^~~~~~~~~~~~~~
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libsyntax/diagnostic.rs:130


Build failed, waiting for other jobs to finish...
Could not compile `objc`.
```

**Comment by tomaka** [comment]
cc @SSheldon 

Probably related to the deprecation of `PhantomFn`.

**Comment by SSheldon** [comment]
Looks like this ICE will be fixed by rust-lang/rust#24056, but I figured out a way to avoid `PhantomFn` and still compile on the beta in SSheldon/rust-objc@ef565e4 and published that as version 0.1.2.

@bvssvni, if you cargo update this should be fixed!
